analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/VerifierBuilder.cs

```diff
@@ -67,14 +67,18 @@ internal record VerifierBuilder
             this with
             {
                 Paths = Paths.Concat(paths).ToImmutableArray(),
-                IsRazor = IsRazor || Array.Exists(paths, x => x.EndsWith(".razor", StringComparison.OrdinalIgnoreCase) || x.EndsWith(".cshtml", StringComparison.OrdinalIgnoreCase))
+                IsRazor = IsRazor || Array.Exists(paths, IsRazorOrCshtmlFile)
             };
 
         public VerifierBuilder AddReferences(IEnumerable<MetadataReference> references) =>
             this with { References = References.Concat(references).ToImmutableArray() };
 
         public VerifierBuilder AddSnippet(string snippet, string fileName = null) =>
-            this with { Snippets = Snippets.Append(new(snippet, fileName)).ToImmutableArray() };
+            this with
+            {
+                Snippets = Snippets.Add(new(snippet, fileName)),
+                IsRazor = IsRazor || IsRazorOrCshtmlFile(fileName)
+            };
```

Comment:
Did you discussed `AddRazorSnippet` method? This approach looks quite hacky, as it's based on filename side-effects.

Reply:
We are checking the extension of the files in a few places for this razor sprint. What do you mean by filename side-effect?

Reply:
The checking for file extension of a snippet is a sideffect from a method contract point of view. And rather hidden way of using it.

Reply:
I see your point. 
I'm not sure when it's convenient to duplicate all those methods (see also [AddPath()](https://github.com/SonarSource/sonar-dotnet/blob/81b0563176aeee773fa8dd2c9f54deccbfe6b3e7/analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/VerifierBuilder.cs#L66)) to avoid setting the isRazor variable. Even if we were to introduce `AddRazorSnippet()` and `AddRazorPath()`, we would still need to validate whether the fileName has the right extension. WDYT?

Reply:
I didn't read all the changes you've done, so I don't know. I'm raising an open question

Reply:
I would keep it this way, but let's wait for @antonioaversa to share his thoughts on this